apps/verticalhome/js/app.js

```diff
@@ -141,6 +156,38 @@
       this.grid.stop();
     },
 
+    getAppByURL: function(url) {
+      return this.itemStore.getAppByURL(url);
+    },
+
+    getPinAppList: function() {
+      return this.itemStore.getPinAppList();
+    },
+
+    savePinAppItem: function(obj) {
+      this.itemStore.savePinAppItem(obj);
+    },
+
+    savePinApps: function(objs) {
+      this.itemStore.savePinApps(objs);
+    },
+
+    showMoreApps: function() {
+      this.inMoreApps = true;
+      console.log("inMoreApps is : " + this.inMoreApps);
+      document.getElementById("main-screen").setAttribute("hidden","");
+      document.getElementById("more-apps-screen").removeAttribute("hidden");
+      MoreAppsNavigation.init();
+      document.getElementsByTagName('gaia-grid-rs')[0].scrollTo(0,0);
+    },
+
+    backToMainScreen: function() {
+      this.__proto__.inMoreApps = false;
```

Comment:
If you ever have to use __proto__ your code is almost certainly wrong. Accessing __proto__ is also a big performance hit, since the code can't be optimized by the JS interpreter.  

You'll have to have a very good reason for __proto__ and a long comment explaining the reason, or I will not land the patch.